### ISSUE TITLE
meson build style prints wrong `Host machine` information

### ISSUE THREAD

**Issue by ericonr** [void-linux/void-packages#31527]
With the same cross file generated by `xbps-src -a armv6l-musl`

In a build inside `xbps-src`, for example `wlroots`:

```
Using 'CFLAGS' from environment with value: '-fstack-clash-protection -D_FORTIFY_SOURCE=2 -O2 -pipe -march=armv6 -mfpu=vfp -mfloat-abi=hard   -I/usr/arm-linux-musleabihf/usr/include'
Using 'LDFLAGS' from environment with value: '-Wl,-z,relro -Wl,-z,now -Wl,--as-needed    -L/usr/arm-linux-musleabihf/usr/lib'
Using 'CPPFLAGS' from environment with value: '  '
C compiler for the host machine: arm-linux-musleabihf-gcc (gcc 10.2.1 "arm-linux-musleabihf-gcc (GCC) 10.2.1 20201203")
C linker for the host machine: arm-linux-musleabihf-gcc ld.bfd 2.35.1
Using 'CC_FOR_BUILD' from environment with value: 'cc'
Using 'CFLAGS_FOR_BUILD' from environment with value: '-O2 -pipe'
Using 'LDFLAGS_FOR_BUILD' from environment with value: ''
Using 'CPPFLAGS_FOR_BUILD' from environment with value: ''
C compiler for the build machine: cc (gcc 10.2.1 "cc (GCC) 10.2.1 20201203")
C linker for the build machine: cc ld.bfd 2.35.1
Build machine cpu family: x86_64
Build machine cpu: x86_64
Host machine cpu family: x86_64
Host machine cpu: x86_64
Target machine cpu family: arm
Target machine cpu: armv6l
```

In a build outside `xbps-src`:

```
C compiler for the host machine: arm-linux-musleabihf-gcc (gcc 10.2.1 "arm-linux-musleabihf-gcc (GCC) 10.2.1 20201203")
C linker for the host machine: arm-linux-musleabihf-gcc ld.bfd 2.35.1
C compiler for the build machine: cc (gcc 10.2.1 "cc (GCC) 10.2.1 20201203")
C linker for the build machine: cc ld.bfd 2.35.1
Build machine cpu family: x86_64
Build machine cpu: x86_64
Host machine cpu family: arm
Host machine cpu: armv6l
Target machine cpu family: arm
Target machine cpu: armv6l
```

It shouldn't be a huge issue and I think things are still working correctly (since it's using the correct compiler), would mostly just like to understand what's wrong.

**Comment by Johnnynator** [comment]
That thing is just a bug that is already fixed upstream. I can't reproduce it with 0.58.1

**Comment by ericonr** [comment]
Yup, fixed with #31529. Thanks!
